src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,246 @@
+            Legal Name: {{ business.name }} <br>
+            Legal Type: {{ business.legalType }} <br>
+            Identifier: {{ business.identifier }} <br>
+            Status: {{ business.status }}
+          </template>
+        </li>
+      </ul>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator'
+import { Action, Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { CommonMixin } from '@/mixins'
+import { BusinessLookupServices, LegalServices } from '@/services'
+import { BusinessLookup } from '@bcrs-shared-components/business-lookup'
+import { BusinessIF, BusinessLookupIF, EmptyBusinessLookup } from '@/interfaces'
+
+@Component({
+  components: {
+    BusinessLookup
+  }
+})
+export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
+  @Getter(useStore) getAmalgamatingBusinesses!: Array<BusinessIF>
+  @Getter(useStore) isRoleStaff!: boolean
+
+  @Action(useStore) setAmalgamatingBusinesses!: (x: Array<BusinessIF>) => void
+
+  // Local properties
+  amalgamatingBusinessesValid = false
+  amalgamatingBuinesses = []
+  initialBusinessLookupObject = EmptyBusinessLookup
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  async setAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
```

Comment:
Can you provide some tests to demonstrate all flows, please?

Reply:
Or, can you show the data from each scenario?

Reply:
Absolutely Sev.

So, for businesses that are in LEAR, here's the data that I'm getting from the lookup:
![big json](https://github.com/bcgov/business-create-ui/assets/122301442/663bf36f-74a4-47e6-a516-5557401a8ea3)

Otherwise, this is what I'm getting:
![small json](https://github.com/bcgov/business-create-ui/assets/122301442/8a707c1b-4d89-406c-9e6c-d306d557d08b)

So, I wanted to handle both flows/scenarios although the ticket mentions to assume all searched businesses are in LEAR.

Reply:
The risk with fallbacks is that they can make it look like things are working normally but something is actually wrong. In this case I think I'd prefer some error handling if the fetched business is not in LEAR.

Reply:
I don't know if we need to use any of the data from the business lookup (except identifier), since we'll get it all from the "fetch business" LEAR call anyway. Maybe one way will make sense as you work on this.